src/SW_netCDF_General.c

```diff
@@ -0,0 +1,1402 @@
+        LogError(
+            LogInfo,
+            LOGERROR,
+            "An error occurred when attempting "
+            "to retrieve the dimension value of "
+            "%d.",
+            dimID
+        );
+    }
+}
+
+/**
+@brief Get a dimension identifier within a given netCDF
+
+@param[in] ncFileID Identifier of the open netCDF file to access
+@param[in] dimName Name of the new dimension
+@param[out] dimID Identifier of the dimension
+@param[out] LogInfo Holds information on warnings and errors
+*/
+void SW_NC_get_dim_identifier(
+    int ncFileID, const char *dimName, int *dimID, LOG_INFO *LogInfo
+) {
+
+    if (nc_inq_dimid(ncFileID, dimName, dimID) != NC_NOERR) {
+        LogError(
+            LogInfo,
+            LOGERROR,
+            "An error occurred attempting to "
+            "retrieve the dimension identifier "
+            "of the dimension %s.",
+            dimName
+        );
+    }
+}
+
+/**
+@brief Check that the constant content is consistent between
+domain.in and a given netCDF file
+
+If ncFileID is negative, then the netCDF fileName will be temporarily
+opened for read-access.
+
+@param[in] SW_Domain Struct of type SW_DOMAIN holding constant
+    temporal/spatial information for a set of simulation runs
+@param[in] ncFileID Identifier of the open netCDF file to check
+@param[in] fileName Name of netCDF file to test (used for error messages)
+@param[out] LogInfo Holds information on warnings and errors
+*/
+void SW_NC_check(
+    SW_DOMAIN *SW_Domain, int ncFileID, const char *fileName, LOG_INFO *LogInfo
+) {
+
+    Bool fileWasClosed = (ncFileID < 0) ? swTRUE : swFALSE;
+
+    /* Get latitude/longitude names that were read-in from domain input file */
+    char *readinYName = SW_Domain->OutDom.netCDFOutput.geo_YAxisName;
+    char *readinXName = SW_Domain->OutDom.netCDFOutput.geo_XAxisName;
+
+    if (fileWasClosed) {
+        // "Once a netCDF dataset is opened, it is referred to by a netCDF ID,
+        // which is a small non-negative integer"
+        if (nc_open(fileName, NC_NOWRITE, &ncFileID) != NC_NOERR) {
+            LogError(
+                LogInfo,
+                LOGERROR,
+                "An error occurred when attempting "
+                "to open the file %s.",
+                fileName
+            );
+            return; // Exit function prematurely due to error
+        }
+    }
+
+    SW_CRS *crs_geogsc = &SW_Domain->OutDom.netCDFOutput.crs_geogsc;
+    SW_CRS *crs_projsc = &SW_Domain->OutDom.netCDFOutput.crs_projsc;
+    Bool geoIsPrimCRS =
+        SW_Domain->OutDom.netCDFOutput.primary_crs_is_geographic;
+    char strAttVal[LARGE_VALUE];
+    double doubleAttVal;
+    const char *geoCRS = "crs_geogsc";
+    const char *projCRS = "crs_projsc";
+    Bool geoCRSExists = SW_NC_varExists(ncFileID, geoCRS);
+    Bool projCRSExists = SW_NC_varExists(ncFileID, projCRS);
+    const char *impliedDomType =
+        (SW_NC_dimExists("site", ncFileID)) ? "s" : "xy";
```

Comment:
here and throughout, please use user-provided name for "site" dimension/variable (domain.in:siteName) instead of hard-coded "site"